Bring back cross-cutting skip_fixtures logic

esoergel opened the pull request:
## Product Description
Reverts the exceptions introduced in https://github.com/dimagi/commcare-hq/pull/35387 and https://github.com/dimagi/commcare-hq/pull/35612
The `skip_fixtures` flag was originally introduced as a performance optimization for faster restores on USH web apps projects that very frequently synced.  However, this meant that fixtures would only be included in the sync on certain types of restores - certain workflows might skip fixture syncing entirely, resulting in a stale sandbox (hence the `ignore_skip_fixtures` hack).  Now instead we're moving towards more targeted logic, where each fixture can decide whether it needs to be updated, based on whether the underlying data has changed since the last sync (eg locations fixture), or based on configured staleness allotments (eg mobile UCR).  Note that as of this writing, lookup tables don't have any smart sync logic - an obvious optimization target.

Separately, https://github.com/dimagi/commcare-android/pull/2680 piggy-backed off this existing flag for a different purpose.  CommCare Android has a "background sync" restore workflow which completely disregards the fixture nodes in the restore, if included.  In this latter case, it is desirable that we do skip fixture logic entirely, as smart syncing logic would be confused by a sync that ignores the restore response.

## Technical Summary
https://dimagi.atlassian.net/browse/USH-5710

## Feature Flag


## Safety Assurance


### Safety story


### Automated test coverage


### QA Plan


### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

esoergel committed:
Bring back cross-cutting skip_fixtures logic

MartinRiese reviewed (approved):
I take it 
```
for provider in providers:
            if self._skip_fixtures and not getattr(provider, 'ignore_skip_fixtures_flag', False):
                continue
```
still exists and this is why this works?

esoergel committed:
lint

esoergel commented:
@MartinRiese I'm not certain what you mean.  I removed the `ignore_skip_fixtures_flag` bit (which is the main change) and moved the `if self._skip_fixtures` check up a level, back where it was prior to https://github.com/dimagi/commcare-hq/pull/35387/files

MartinRiese reviewed (approved):
I meant https://github.com/dimagi/commcare-hq/blob/22fb9e3b22b254463be2026243a966b5ec8c484b/corehq/ex-submodules/casexml/apps/phone/fixtures.py#L60, which uses `ignore_skip_fixtures_flag`
